### PR TITLE
build oci-images also on head-update

### PR DESCRIPTION
**Pull request by ccwienk**

### CODE REVIEW

**Comment by ocm-ci-robot-0** [comment]
This PR proposes changes that would break the pipeline definition:
```
delivery-service-build-images-on-headupdate: Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 141, in render
    definition_descriptor = self._render(definition_descriptor)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/replicator.py", line 185, in _render
    'definition': factory.create_pipeline_definition(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 88, in create_pipeline_definition
    self._apply_traits(variant)
  File "/usr/lib/python3.12/site-packages/concourse/factory.py", line 177, in _apply_traits
    transformer.process_pipeline_args(pipeline_def)
  File "/usr/lib/python3.12/site-packages/concourse/model/traits/publish.py", line 749, in process_pipeline_args
    input_step = pipeline_args.step(input_step_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/concourse/model/job.py", line 238, in step
    raise ValueError(f'no such step: {name=}')
ValueError: no such step: name='build_python_packages'

```

**Comment by ocm-ci-robot-0** [comment]
This PR proposes changes that would break the pipeline definition:
```
<invalid YAML>: mapping values are not allowed here
  in "<unicode string>", line 160, column 30:
            build_python_packages:
                                 ^
```

**Comment by ocm-ci-robot-0** [comment]
The pipeline-definition has been fixed.

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -164,3 +167,14 @@ delivery-service:
               triggering_policy: only_first
         component_descriptor:
           ocm_repository: europe-docker.pkg.dev/gardener-project/snapshots
+        publish:
+          dockerimages:
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply .ci/pipeline_definitions:171]: you are, of course, correct m(

**Pull request merged**